src/util.py

```diff
@@ -371,3 +372,15 @@ def make_required_dirs(path: str):
     """
     out_path = Path(path).parent
     out_path.mkdir(parents=True, exist_ok=True)
+
+def add_rank_column(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a column to the dataframe that ranks the rows by the group_cols.
```

Comment:
This doesn't describe the current behavior. The implementation doesn't use a `group_cols` variable and adds a default ranking of 1, which is what we wanted.